test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/BuildIncrementalismTest.cs

```diff
@@ -136,5 +137,46 @@ public BuildIncrementalismTest(BuildServerTestFixture buildServer)
                 Assert.FileDoesNotExist(result, IntermediateOutputPath, "Razor", "Views", "Home", "Index.cshtml.g.cs");
             }
         }
+
+        [Fact]
+        [InitializeTestProject("ClassLibrary")]
+        public async Task Build_TouchesUpToDateMarkerFile()
+        {
+            var classLibraryDll = Path.Combine(IntermediateOutputPath, "ClassLibrary.dll");
+            var classLibraryViewsDll = Path.Combine(IntermediateOutputPath, "ClassLibrary.Views.dll");
+            var markerFile = Path.Combine(IntermediateOutputPath, "ClassLibrary.csproj.CopyComplete");
+
+            var result = await DotnetMSBuild("Build");
+            Assert.BuildPassed(result);
+
+            Assert.FileExists(result, classLibraryDll);
+            Assert.FileExists(result, classLibraryViewsDll);
+            Assert.FileExists(result, markerFile);
+
+            // Gather thumbprints before incremental build.
+            var classLibraryThumbPrint = GetThumbPrint(classLibraryDll);
+            var classLibraryViewsThumbPrint = GetThumbPrint(classLibraryViewsDll);
+            var markerFileThumbPrint = GetThumbPrint(markerFile);
+
+            result = await DotnetMSBuild("Build");
+            Assert.BuildPassed(result);
+
+            // Verify thumbprint file is unchanged between true incremental builds
+            Assert.Equal(classLibraryThumbPrint, GetThumbPrint(classLibraryDll));
+            Assert.Equal(classLibraryViewsThumbPrint, GetThumbPrint(classLibraryViewsDll));
+            // In practice, this should remain unchanged. However, since our tests reference
+            // binaries from other projects, this file gets updated by Microsoft.Common.targets
+            Assert.NotEqual(markerFileThumbPrint, GetThumbPrint(markerFile));
```

Comment:
This is kinda crummy. Microsoft.Common.targets always updates the file because we reference binaries inside tests. I verified this in a vanilla project and things worked correctly. Regardless, keeping this file in so we can ensure that updating just the views dll results in the thumbprint being updated.